### PR TITLE
#1105 Added checks on timelinecells, so scrub is tracked

### PR DESCRIPTION
**Pull request by davidlamhauge**
This PR see to it, that the scrubber always is visible during playback.
It is a rather simple solution, since we shouldn't do too complicated things on the timeline, while we wait for the Timeline rewrite.
This one has bugged me (and @Jose-Moreno) and possible others for some time, so here is a solution.
1105_scrubPlaybackTracking

### CODE REVIEW

File: core_lib/src/interface/timelinecells.cpp [hunk 1]
```diff
@@ -425,8 +425,20 @@ void TimeLineCells::paintEvent(QPaintEvent*)
         }
 
         // --- draw the position of the current frame
```
Comment (CandyFace) [inline core_lib/src/interface/timelinecells.cpp]: Please update or move the comment so it matches the behaviour context.

File: core_lib/src/interface/timelinecells.cpp [hunk 2]
```diff
@@ -425,8 +425,20 @@ void TimeLineCells::paintEvent(QPaintEvent*)
...
-        if (mEditor->currentFrame() > mFrameOffset)
+        if (mEditor->playback()->isPlaying() && mEditor->currentFrame() < mFrameOffset)
         {
+            mFrameOffset = mEditor->currentFrame() - 1;
+            emit offsetChanged(mFrameOffset);
+            mTimeLine->updateContent();
+        }
+        else
+        {
+            if (mEditor->playback()->isPlaying() && width() < (mEditor->currentFrame() - mFrameOffset + 1) * mFrameSize)
```
Comment (CandyFace) [inline core_lib/src/interface/timelinecells.cpp]: The code here could be tied up by putting it into a separate function, calling it trackScrubber() or something similar.

Comment (davidlamhauge) [reply core_lib/src/interface/timelinecells.cpp]: Moved tracking to separate function. It seem to work like before.

**Comment by CandyFace** [comment]
Since the number of changes here are small, we can merge this right away, however to give the chance of someone else to review it, I'll wait till tomorrow before merging.

**Comment by davidlamhauge** [comment]
It is indeed small changes, but let it rest a day if someone else wants to review it. Thanks!

**Comment by CandyFace** [comment]
I agree that would be useful.

**Comment by davidlamhauge** [comment]
Two small errors fixed.
When moving left, the scrubber went out of sight.
If you used the horizontal scrollbar to move the scrubber out of sight, and then pressed Play, the scrubber would remain out of sight, until it reached the place you scrolled to.
Now everything should work as expected, whether you Play, Scrub the timeline, or move to previous/next frame or keyframe.
Feel free to test it!

**Comment by davidlamhauge** [comment]
I have made so it behave likewise forwards and backwards.
I don't understand the last paragraph about pausing the scrubber, so I haven't addressed that.
I meant this PR to be a quick-fix for an old problem, while we're waiting for the new timeline, but I'll of course be ready to make more changes, if they are wanted.

**Comment by davidlamhauge** [comment]
It is less timeline updates, that's the reason for jumping halfway back, when it goes of screen. I thought that it would be too much if it went all the way over to the left.
But - I am ok with letting it jump to the far left, if we agree on that.

**Comment by scribblemaniac** [comment]
@CandyFace Do you have any comments on the discussion about how far back the scrubber should go during playback?

**Comment by CandyFace** [comment]
> @CandyFace Do you have any comments on the discussion about how far back the scrubber should go during playback?

Not really, I think it's fine to scrub to the middle of the view.

**Comment by CandyFace** [comment]
Two developers has approved the changes, so I'm merging. If there's a need to further discuss the functionality, a new issue can be created.

**Pull request merged**